runtime-common/realm.ts

```diff
@@ -99,16 +96,28 @@ export class Realm {
       this.#adapter.readdir.bind(this.#adapter),
       this.readFileAsText.bind(this)
     );
-    this.#loader = new Loader(this, async (url: URL) => {
-      return this.transpileJS((await this.getCardSourceAsText(url))!, url.href);
-    });
+    this.loader = Loader.getLoader(
+      new Map([
+        [
+          this.url,
+          async (url: URL) => {
+            return this.transpileJS(
+              (await this.getCardSourceAsText(url))!,
+              url.href
+            );
+          },
+        ],
+      ])
+    );
```

Comment:
As realms are created, they will supply a file loader for the paths they govern to the single global loader, such that the loader will be able to use file access as opposed to fetch access for any URL's that are governed by realms running on the server